src/orders/service/order.service.ts

```diff
@@ -0,0 +1,92 @@
+import { BadRequestException, ForbiddenException, Injectable, NotFoundException } from "@nestjs/common";
+import { UUID } from "crypto";
+import { DeleteResult } from "typeorm";
+import { OrderRepository } from "../repository/order.repository";
+import { Order } from "../domain/order.entity";
+import { OrderItem } from "../dto/create-order.dto";
+import { StockService } from "src/products/service/stock.service";
+import { Stock } from "src/products/domain/stock.entity";
+import { OrderDetail } from "../domain/order-detail.entity";
+import { OrderDetailService } from "./order-detail.service";
+
+@Injectable()
+export class OrderService{
+
+    constructor(
+        private readonly orderRepository: OrderRepository,
+        private readonly orderDetailService: OrderDetailService,
+        private readonly stockService: StockService
+    ){}
+
+    async getAll(): Promise<Order[]>{
+        return await this.orderRepository.getAll();
+    }
+
+    async getById(id: UUID): Promise<Order>{
+        try {
+            return await this.orderRepository.getById(id);
+        } catch (error) {
+            throw new NotFoundException("Order not found for ID: " + id);
+        }
+    }
+
+    async add(order: Order, orderItems: OrderItem[]): Promise<Order>{
+        const stock= await this.stockService.get(orderItems);
+        const {notInStock, updatedStock} = this.checkStock(orderItems, stock);
+        if(notInStock.length>0)
```

Comment:
**suggestion(non-blocking):** note that some parts of this file are not formatted, as a future reference there are different ways to setup formatting (e.g.: [prettier](https://prettier.io/))

Reply:
I was already using prettier, but did not globally format the project.
Solved using: `npm run format`.